app/Http/Controllers/Manager/WalletController.php

```diff
@@ -328,6 +329,114 @@ public function withdraw(Request $request, AdsRpcClient $rpcClient): JsonRespons
         return $this->withdrawAds($request, $rpcClient);
     }
 
+    private function getSwashVaultBSCAddress(): WalletAddress
+    {
+        try {
+            return new WalletAddress('BSC', config('app.svault_bsc_address'));
+        } catch (InvalidArgumentException $e) {
+            Log::error(sprintf('Invalid BSC address is set: %s', $e->getMessage()));
+            throw new HttpException(Response::HTTP_INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    public function withdrawBatchSwash(Request $request, AdsRpcClient $rpcClient): JsonResponse
+    {
+        /** @var User $user */
+        $user = Auth::user();
+
+        if(!$user->isModerator()){
+            return self::json(['Auth' => 'Only Moderator/Admin can run this API'], JsonResponse::HTTP_FORBIDDEN);
+        }
+
+        $addressFrom = $this->getAdServerAdsAddress();
+        $address = $this->getSwashVaultBSCAddress();
+        $addressTo = $this->getWalletAdsAddress($rpcClient, $address);
+        $message = $this->getWalletAdsMessage($rpcClient, $address);
+        $batchId = hash('sha256', strval(microtime(true)) );
+
+        $balanceTotal = UserLedgerEntry::getWalletBalanceForAllUsers();
+        $users_balances = UserLedgerEntry::allWalletBalanceIfAny();
+
+        $amount = AdsUtils::calculateAmount($addressFrom, $addressTo, $balanceTotal);
+        $adsFee = AdsUtils::calculateFee($addressFrom, $addressTo, $amount);
+        $total = $amount + $adsFee;
+
+        if ($balanceTotal < $total) {
+            throw new UnprocessableEntityHttpException(sprintf('Insufficient total: %d, fee: %d', $balanceTotal, $adsFee));
+        }
+        DB::beginTransaction();
+        
+        foreach ($users_balances as $item) {
+            $user_item = User::fetchById($item['uid']);
+            $ledgerEntry = UserLedgerEntry::constructSwash(
+                $batchId,
+                $user_item->id,
+                -$item['share']
+            )->addressed($addressFrom, $addressTo);
+            if (!$ledgerEntry->save()) {
+                DB::rollBack();
+                throw new InternalErrorException();
+            }
+        }
+        DB::commit();
+        $logMessage = '[WalletController] Dispatching AdsSendOneSwash with batchId (%s) addressTo (%s) amount (%s) message (%s) users_balances (%s).';
+        Log::info(sprintf($logMessage, $batchId, $addressTo, $amount, $message, json_encode($users_balances)));
```

Comment:
In the production, `info` logs normally are omitted. So, this logging makes no sense.